File: app/src/main/java/com/example/iteventscheckin/presentation/events/EventsViewHolder.java
package com.example.iteventscheckin.presentation.events;

import android.view.View;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.example.iteventscheckin.room.Event;
import com.example.iteventscheckin.presentation.OnItemListClickListener;
import com.example.iteventscheckin.R;

public class EventsViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
    private TextView titleView;

    private TextView descView;

    private TextView dateView;

    private OnItemListClickListener listener;

    public EventsViewHolder(@NonNull View itemView, OnItemListClickListener listener) {
        super(itemView);
        this.listener = listener;
        titleView = itemView.findViewById(R.id.eventTitle);
        descView = itemView.findViewById(R.id.eventDesc);
        dateView = itemView.findViewById(R.id.eventDate);
        itemView.setOnClickListener(this);
    }

    public void bind(Event model) {
        titleView.setText(model.getTitle());
        descView.setText(model.getDescription());
        dateView.setText("TODO:");
    }

    @Override
    public void onClick(View v) {
        listener.onItemListClick(getAdapterPosition());
    }
}

File: app/src/main/java/com/example/iteventscheckin/presentation/events/EventsViewHolder.kt
package com.example.iteventscheckin.presentation.events

import android.view.View
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.iteventscheckin.DateUtils
import com.example.iteventscheckin.models.Event
import com.example.iteventscheckin.presentation.OnItemListClickListener
import com.example.iteventscheckin.R

class EventsViewHolder(itemView: View, private val listener: OnItemListClickListener) :
    RecyclerView.ViewHolder(itemView), View.OnClickListener {
    private val titleView: TextView

    private val descView: TextView

    private val dateView: TextView

    init {
        titleView = itemView.findViewById(R.id.eventTitle)
        descView = itemView.findViewById(R.id.eventDesc)
        dateView = itemView.findViewById(R.id.eventDate)
        itemView.setOnClickListener(this)
    }

    fun bind(model: Event) {
        titleView.text = model.title
        descView.text = model.description
        dateView.text = String.format("%s - %s", DateUtils.dateFromFormat(model.date?.start),
            DateUtils.dateFromFormat(model.date?.end))
    }

    override fun onClick(v: View) {
        listener.onItemListClick(adapterPosition)
    }
}
